config/configuration_service.proto

```diff
@@ -7,65 +7,128 @@ package config;
+    }
 
     //Restores the normal behavior to start processing new requests again
     // ("_Request_Start", "block_number",authentication_address) should be sent in the signature
-    rpc StartProcessingRequests(CommandRequest) returns (Response) {}
-}
+    rpc StartProcessingRequests (CommandRequest) returns (Response) {
+    }
+
+    //To Determine if the Daemon is processing requests are not.
+    // ("__Is_Daemon_Processing_Requests", "block_number",authentication_address) should be sent in the signature
+    rpc IsDaemonProcessingRequests (ReadRequest) returns (Response) {
+    }
+
 
 
+}
+
 //Used when you just need to read the current configuration
 message ReadRequest {
     //Signature will compromise of the below
     // ("_Request_Read", "block_number",authentication_address)
+    //OR
+    //("__Is_Daemon_Processing_Requests", "block_number",authentication_address)
     bytes signature = 1;
 
+    //current block number (signature will be valid only for short time around this block number)
+    uint64 current_block = 2;
+
+    //The address of the person who has rights to make configuration changes on Daemon
+    string user_address = 4;
+
 }
 
+
+
 //Used when you want to update the existing configuration
 message UpdateRequest {
     //Signature will compromise of the below
-    // ("_Request_Update", "block_number","updated_json_data",authentication_address)
+    // ("_Request_Update", "block_number",authentication_address)
     bytes signature = 1;
-    //Holds the updated configuration in the form of JSON,only the fields to be updated should be sent
-    string updated_json_data = 2;
+    //Indicates the updated configuration ( only the modified leaf and its changed value is passed ) Example (log.output.rotation_time_in_sec is a valid key )
+    map<string, string> updated_configuration = 2;
 
-}
+    //current block number (signature will be valid only for short time around this block number)
+    uint64 current_block = 3;
+
+    //The address of the person who has rights to make configuration changes on Daemon
+    string user_address = 4;
 
+}
 
 message CommandRequest {
     //Signature will compromise one of the below
     // ("_Request_Start", "block_number",authentication_address) for the StartProcessingRequests
     // ("_Request_Stop", "block_number",authentication_address) for the StopProcessingRequests
     bytes signature = 1;
+    //current block number (signature will be valid only for short time around this block number)
+    uint64 current_block = 2;
+    //The address of the person who has rights to make configuration changes on Daemon
+    string user_address = 3;
 }
 
 
-
 message Response {
     //Signature will compromise of the below
     //(block_number,authentication_address)
     bytes signature = 1;
+    //Indicates the current processing status ( true means Daemon is processing and false means it is not processing any requests)
+    bool current_processing_status = 2;
```

Comment:
1. `Response` sounds general, what do you think about `StatusResponse`
2. I would replace `bool` by enum here, it would make code more self-descriptive:
```proto
message StatusResponse {
    //Signature will compromise of the below
    //(block_number,authentication_address,status)
    bytes signature = 1;

    enum Status {
    	// Daemon has no request in progress
    	IDLE = 0;
    	// Daemon is processing request
    	REQUEST_IN_PROGRESS = 1;
    }

    //Indicates the current processing status
    Status current_processing_status = 2;
}
```

Reply:
Ah, So the original thought was to know if the Demon has stopped processing any request ( speiclaly after the StopProcessingRequests / StartProcessingRequests are called ) 
But I like this better , as we can adopt to more states of Daemon in to future , but have something like HAS_STOPPED_PROCESSING_REQUESTS , IS_READY_TO_PROCESS_REQUESTS ?in the enum

Reply:
Yes, one of the enum's advantages is that it can be extended in future.

Reply:
I don't insist on specific names I proposed, main idea is to have such self-descriptive names.

Reply:
Ah ok, I see that you have added `HAS_STOPPED_PROCESSING_REQUESTS` already - good idea :+1: